### DISCUSSION TITLE
封測前測試

### DISCUSSION THREAD

**Discussion by MROS** [carbon-bond/carbonbond#283] [Show and tell]
# 碳鍵已有功能總覽暨測試指南

#### 右上主選單
![](https://i.imgur.com/o4pfEzw.png)

#### 左側菜單
![](https://i.imgur.com/f8yjNNj.png)
由上到下，分別爲
- 瀏覽
- 聊天室
- 交友
- 草稿

#### 文章功能列
![](https://i.imgur.com/2N8Zjvv.png)

## 帳號

1. 註冊
    1. 使用律師證號/姓名搜尋到律師
    1. 收到邀請信
    1. 進入邀請信中的鏈接輸入帳號密碼
    1. 輸入帳號密碼後，點入原本邀請信鏈接，應該已失效
1. 登入
    1. 輸入帳號密碼登入
    1. 重新整理後，依然處於登入狀態
1. 登出
    1. 重新整理後依然處於登出狀態
    1. 在登出狀態中，左側菜單中
        1. 應該只剩下瀏覽
        1. 瀏覽功能中，只剩下
            1. 特化瀏覽
                1. 全站熱門
                1. 所有看板
            1. 熱門看板
    1. 請回報所有你認爲不應該出現在未登入狀態下的東西
1. 重設密碼
    1. 在右上主選單 -> 設定
    1. 收到重設密碼信件
    1. 進入信件中超鏈接重設密碼
    1. 重設密碼後，回到原本超鏈接，應該已失效

## 政黨
點擊右上主選單「我的政黨」
1. 可以創建政黨
1. 可以看到以前創立的政黨

## 看板
我的政黨 => 點擊任意政黨 => 創立看板
1. 嘗試創立看板
1. 看板定義是否足夠直觀？
1. 看板定義的範本是否可用？
1. 嘗試創建擁有多個分類、多個欄位、不同種類型的欄位的看板

## 文章
創立完看板後，可嘗試發表文章

1. 發表
   1. 嘗試在不同分類、不同欄位中以不同內容發文
   1. 在多行文本中，留下含有網址／圖片網址的內容 
1. 觀看
   1. 文章發表後，是否出現在
       1. 所屬看板
       1. 我的卷宗 => 文章
   1. 網址、圖片網址是否能正常顯示
1. 匿名
   1. 嘗試發表匿名文章
   1. 觀察匿名文章是否能被小號看到作者

### 收藏
1. 點選文章工具列的「收藏」
1. 進入右上主選單「我的卷宗」=>「收藏」，看是否已經收藏

## 留言
1. 進入任意一篇文章，發起留言
1. 檢查留言數量是否正確
1. 留言中含有換行（shift + enter 可以換行）/網址/圖片網址時能正常顯示

## 我的卷宗
1. 設置一句話介紹
1. 換頭貼
1. 更改自我介紹
1. 檢查已認證稱號

## 人際關係
1. 進入他人卷宗，點擊
    1. 公開追蹤
    1. 公開仇視
    1. 偷偷追蹤
    1. 偷偷仇視

觀察自己以及此人的被追蹤／仇視數量是否正確、「偷偷」是不是真的偷偷

## 個版
1. 點擊右上主選單 => 我的個版
1. 檢查是否能成功建立各版

## 瀏覽功能
1. 我的追蹤
    1. 點擊任意文章功能列的「追蹤」
1. 全站熱門
1. 所有看板
    1. 創版，並觀察新創的版是否出現在所有看板
3. 熱門看板
4. 訂閱看板
    1. 進入任一看板，點擊「訂閱看板」，應該能在左側菜單 => 訂閱看板中看到該看板

## 聊天功能
1. 建立新聊天室
    1. 進入小號的卷宗，點擊私訊，應該要在底部出現一個聊天室
    1. 輸入訊息
    1. 重新整理，應該要能在左側菜單 => 聊天中看到該聊天訊息仍在
    1. 登入小號，應該要在左側菜單=>聊天中看到一新聊天訊息
1. 持續聊天
    1. 與小號聊天一段時間（斷斷續續，時聊時不聊），觀察是否會斷線
    1. 重新整理，觀察是否有漏訊息

## 草稿功能
1. 進入看板，點出編輯器

## 通知功能
1. 調正對小號的人際關係，小號是否收到通知？
1. 對小號的文章留言，小號是否收到通知？
1. 回覆小號的文章，小號是否收到通知？

**Comment by ri31625** [comment]
1.左側菜單：**訊息icon**在行動裝置上無法顯示，🗐 的 emoji 太少字體支援，需尋求替代符號。
2.**我的個版**無法建立。
3.Header看板名稱無法直接連結到該看板（建立超連結）。例如，當我瀏覽八尬板上的一篇文章時，點擊 header 上的看板名稱，並沒有辦法把我再帶到看板頁去。
4.文章頁要顯示該篇文章連結到的文章。
5.個別聊天室的**齒輪**目前無功能，建議刪除。
6.**全站熱門**不會顯示。
7.我的訂閱改為「**我的追蹤**」。
8.與人建立關係後，贊同、反對之回覆文無法跳出通知。發文時還會 toast 一大塊錯誤訊息。
9.**鍵能**不會顯示數字。
10.**分享**：複製網址。
11.每隔一段時間會自動登出。
12.文章半徑目前每篇都是一樣的
13.我的政黨加入一篇說明構想的介紹（暫不啟用）。
14.在線人數改為「**本日發文數量**」。
15.全站搜尋選單刪除，改為「**標題搜尋：_________________**」。
16.文章頁右邊「關於作者」放入「**我的卷宗**」之中的作者資訊；「廣告」放入「**招租中，意者請洽：公務用gmail信箱**」。

**Reply by MROS** [reply]
還有手機版目前功能太少，至少要能支援
- 看文
- 留言
- 註冊、登入
- 找回密碼
- 發文
- 聊天

**Comment by fro11o** [comment]
- 右上角下拉選單 設定 emoji 跟其他的大小似乎不同
- Mac 有些 icon 無法顯示  (e.g. 鉛筆，左側瀏覽) 
- 我的政黨介面 UI 是否不明確
- 登出後畫面會停留在原地，應該要回到首頁之類的
- 文章編輯器縮小後登出，不會消失
- 現在不登入好像也可以看到看板列表，也可以點進去看到文章，是預期中的？
- 是否需要考慮註冊功能可能會被濫用，ex 一直幫別人申請，或是被查到 api 然後寫腳本一直送
- 點進文章頁面後，右側關於作者沒有內容
- 我的訂閱，是否會跟訂閱看板混淆
- 無法建立我的個版
- 小號留言，原PO沒有通知
